### PR TITLE
Fixing Check-In duration when stopover line is a ring

### LINKED ISSUE #37: Negative trip times
**Describe the bug**
https://traewelling.de/status/129666 (which also seems to suffer from the bug described in  https://github.com/Traewelling/traewelling/issues/10) and https://traewelling.de/status/129656 are trips with a negative journey time. The bus/train also seems to arrive at the destination before it leaves at the start.

**To Reproduce**
1. Search for `Höhenstr., Potsdam`
2. Select an 603 bus to `Platz der Einheit/West, Potsdam`
3. Check-in until `Rathaus, Potsdam` (any other stop might also work)
4. Check the logged journey time as well as the arrival and departure times

Alternatively:
1. Search for `Berlin Westkreuz`
2. Select an S 42 train to `Berlin Südkreuz (S)`
3. Check-in until `Berlin-Tempelhof` (any other stop after that might also work)
4. Check the logged journey time as well as the arrival and departure times

**Expected behavior**
The logged journey time should not be negative and the bus/train should not arrive before it has started. Also, the arrival time should be correct (the departure time is already correct).

**Additional context**
The bug seems to happen when two busses (one-way and back) or two trains (two times around the circle) are squeezed into a single hafas trip. When this happens, stations can appear multiple times in one trip. It seems like Traewelling then internally chooses the first time a station appears in the trip as the destination, rather than that one the user really chose.

### PR DESCRIPTION
**Pull request by MrKrisKrisu**
Child PR from #38 
This is forked from `dev/jv/37-negative-trip-times`

fixes #37

### CODE REVIEW

File: app/Models/TrainCheckin.php [hunk 1]
```diff
@@ -93,12 +94,15 @@ public function getMapLines() {
...
-    }
+    public function getDurationAttribute(): int {
+        try {
+            return $this->origin_stopover->departure_planned->diffInMinutes(
+                $this->destination_stopover->arrival_planned
+            );
+        } catch (Exception) {
+            //We need the try-catch to support old checkins, where no stopovers are saved.
+            return $this->arrival->diffInMinutes($this->departure);
+        }
```
Comment (jeyemwey) [inline app/Models/TrainCheckin.php]: Also DA hättest du dir auch denken können, dass die Codacy-Götter das nicht durchlassen.

Comment (MrKrisKrisu) [reply app/Models/TrainCheckin.php]: Sorry. xD

**Comment by aledjones** [comment]
Finally! 🎉

**Comment by HerrLevin** [comment]
Since the new business logic now requires a supplied departure and arrival time in the check in process, ring lines will still be broken in the API. This needs to be fixed in a future™ update b/c Kris despises the API and doesn't want to work with it.

**Comment by MrKrisKrisu** [comment]
> Since the new business logic now requires a supplied departure and arrival time in the check in process, ring lines will still be broken in the API. This needs to be fixed in a future™ update b/c Kris despises the API and doesn't want to work with it.

I don't despise it, I just don't support it.

**Review by HerrLevin — APPROVED** [review]

**Pull request merged**